controller/scrape.go

```diff
@@ -40,66 +39,79 @@ var numericStatus = map[string]float64{
+	ClusterFailedExperiments.WithLabelValues().Set(fail)
+	ClusterTotalExperiments.WithLabelValues().Set(total)
+}
+func setEngineChaosMetrics(engineDetails ChaosEngineDetail) {
+	EngineTotalExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.TotalExp)
+	EnginePassedExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.PassedExp)
+	EngineFailedExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.FailedExp)
 }
 
-//setChaosResultValue will populate the default value of chaos result
-func setChaosResultValue(clientSet *clientV1alpha1.Clientset, chaosResultSpec ChaosResultSpec) map[string]string {
-	chaosResultMap := make(map[string]string)
-	for _, test := range chaosResultSpec.ChaosExperimentList {
-		chaosResultName := fmt.Sprintf("%s-%s", chaosResultSpec.ExporterSpec.ChaosEngine, test)
-		testResultDump, err := clientSet.LitmuschaosV1alpha1().ChaosResults(chaosResultSpec.ExporterSpec.AppNS).Get(chaosResultName, metav1.GetOptions{})
-		if err != nil {
-			if strings.Contains(err.Error(), "not found") {
-				// lack of result cr indicates experiment not executed
-				chaosResultMap[test] = "not-executed"
-			}
-			continue
+func getExperimentMetricsFromEngine(chaosEngine *litmuschaosv1alpha1.ChaosEngine) (float64, float64, float64) {
+	var total, passed, failed float64
+	passed = 0
+	failed = 0
+	expStatusList := chaosEngine.Status.Experiments
+	total = float64(len(expStatusList))
+	for _, v := range expStatusList {
+		verdictFloat := getValueFromVerdict(v.Verdict)
+		if verdictFloat == 3 {
+			passed++
+		} else if verdictFloat == 2 {
+			failed++
+		} else if verdictFloat == 1 {
+			defineRunningExperimentMetric(chaosEngine.Name, chaosEngine.Namespace, v.Name)
 		}
-		chaosResultMap[test] = testResultDump.Status.ExperimentStatus.Verdict
 	}
-	return chaosResultMap
+	return total, passed, failed
+
 }
+func defineRunningExperimentMetric(engineName string, engineNamespace string, experimentName string) {
+	log.Printf("Running Experiment Metrics: EnginaName: %v, EngineNamespace: %v, ExperimentName: %v", engineName, engineNamespace, experimentName)
+	RunningExperiment.WithLabelValues(engineNamespace, engineName, experimentName, fmt.Sprintf("%s-%s", engineName, experimentName)).Set(float64(1))
 
-// calculateChaosResult will calculate the number of pass and failed experiments
-func calculateChaosResult(chaosResult map[string]string) (ChaosExpResult, map[string]float64) {
-	var cr ChaosExpResult
-	StatusMap := make(map[string]float64)
-	for index, status := range chaosResult {
-		if status == StatusPassed {
-			cr.TotalPassedExp++
-		} else if status == StatusFailed {
-			cr.TotalFailedExp++
-		}
-		StatusMap[index] = statusConversion(status)
-	}
+}
 
-	return cr, StatusMap
+func getValueFromVerdict(verdict string) float64 {
+	if verdict == "Pass" || verdict == "pass" || verdict == "passed" || verdict == "Passed" {
```

Comment:
convert the verdict in small case then match, Then only two case we have to cover